portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -29,34 +29,48 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-	private List<String> comments;
+	private List<Comment> comments;
 
  	@Override
     public void init() {
         comments = new ArrayList<>();
-
-        comments.add("Your portfolio is amazing!");
-        comments.add("It looks good, could be better");
-        comments.add("Add more content!");
-        comments.add("I love the design, but try to add more info");
     }
 
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        String author = "me";
-        Date date = new Date();
+        // creates json string from comments list
+    	// TODO: use GSON instead   
+        String json = "{ \"comments\": [";
+
+        for (int i = 0 ; i < comments.size(); i++) {
+            json += convertToJson(comments.get(i));
+            if (i != comments.size() - 1) {
+                json += ",";
+            }
+       }
+
+        json += "] }";
+
+        // Send the JSON as the response
+        response.setContentType("application/json;");
+        response.getWriter().println(json);
+    }
 
-        for (String commentary: comments) {
-            // Create comment object and convert to json string
-            Comment comment = new Comment(commentary, author, date);
-            String jsonComment = convertToJson(comment);
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        // Get the input from the form.
+        String author = getParameter(request, "author", "unknown");
+        String text = getParameter(request, "text", "");
 
-        	// Send the JSON as the response
-            response.setContentType("application/json;");
-        	response.getWriter().println(jsonComment);
-        }
+        // creates new comment object and puts into list
```

Comment:
Generally it is ok to leave a comment like this out, the line below explains itself. 

Usually we try to add comments about why something is done as opposed to what is happening. If we have to comment and what is happening that is also a sign that the code is complex enough that it belongs in its own function with a descriptive name.

Reply:
Just to make sure I'm not sounding confusing, I am talking about just the code snippets here. For a comment on top of a function it's the opposite. There the comments should say what the function does, as opposed to how or why it should be used.